### DISCUSSION TITLE
Swagger Parser only supports versions 3.0.0, 3.0.1, 3.0.2

### DISCUSSION THREAD

**Discussion by shavo007** [Kong/insomnia#4352] [Ideas] [answered]
**Describe the bug**
Swagger Parser only supports versions 3.0.0, 3.0.1, 3.0.2

Insomnia iso cli does not support the latest version 3.0.3

**To Reproduce**

Example spec

```yaml
openapi: 3.0.3
info:
  version: 1.0.0
  title: Greeting API
  termsOfService: 'https://example.com/terms/'
  contact:
    email: contact@example.com
    url: 'http://example.com/contact'
  license:
    name: Apache 2.0
    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
```

```bash
inso generate config ../oas.yaml --type declarative > ./kong.yaml

```

**Expected behavior**
gen declarative config for kong

**Desktop (please complete the following information):**
 - OS: [e.g. macOS 10.15.1] mac
 - Installation Method: [e.g. homebrew] node `npm i -g insomnia-inso`
 - App Version [e.g. v7.0.3] 2.2.28

**Additional context**

**Comment by apisaint** [comment]
Looks like `openapi-2-kong` relies on `8.x` of `swagger-parser` moving to `10.x` should resolve the issue. Looking to see if this would cause any issues, should only cause issue if the yaml export feature is used.

*Sidenote:* There is also the `insomnia-importers` package which should also upgrade `swagger-parser` as well, it's currently on `6.x` and bumping to `10.x` posed no problems there.

---

As a note, you should also always provide the `paths` property:

```yaml
openapi: 3.0.3
info:
  version: 1.0.0
  title: Greeting API
  termsOfService: 'https://example.com/terms/'
  contact:
    email: contact@example.com
    url: 'http://example.com/contact'
  license:
    name: Apache 2.0
    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
paths: {}
```

**Comment by apisaint** [comment]
Have it working, it will also require having the `servers` property filled out. Going to push up a PR now.

**Comment by shavo007** [comment]
> Have it working, it will also require having the `servers` property filled out. Going to push up a PR now.

👍

Don't take my example as full-blown. It was only to showcase the version!

**Comment by rLitto** [comment]
Also by now there is version 3.1.0 available
Any progress in this?

**Comment by rLitto** [comment]
any news on this?

**Comment by wdawson** [comment]
Hey folks, I've moved this to a GitHub Discussion so we can better talk about various elements of this and better gauge interest. 

As for an update, we are grateful for [the PR that was submitted last year for 3.0.3](https://github.com/Kong/insomnia/pull/2984) and would love to merge it in as quickly as we can. It's a matter of time and priorities. We're trying to be better about merging PRs in a more timely manner and are going to start experimenting with various things to do that later this month. Thanks for your patience on it so far.

As for 3.1.0, it's more like a 4.0 (OpenAPI stopped using SemVer), so that will take a lot more time to support, I'll wager.

**Comment by filfreire** [comment, marked as answer]
In latest Insomnia releases both 3.0.3 and 3.1 are now supported.
